pkg/client/hadialer/hadialer_test.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hadialer
+
+import (
+	"slices"
+	"testing"
+)
+
+func TestInterleaveIPV4AndV6(t *testing.T) {
+	addrs := []string{
+		"192.168.200.1",
+		"10.111.3.1",
+		"192.168.200.2",
+		"fd11::102",
+		"fd11::101",
+	}
+
+	interleaveIPV4AndV6(&addrs)
+
+	expected := []string{
+		"192.168.200.1",
+		"fd11::102",
+		"10.111.3.1",
+		"fd11::101",
+		"192.168.200.2",
```

Comment:
@mcluseau 

Does it make sense to also test with scoped IPv6 addresses here? So e.g.

`fe80::1%eth0`
`fe80::2%eth0`
`fe80::1%eth1`

* RFC if you fancy: https://datatracker.ietf.org/doc/html/rfc4007#section-6
* or if those were URLs: https://datatracker.ietf.org/doc/html/rfc6874

I am wondering if this was even tested (filtered if not allowed) prior to this PR?

Reply:
ummm not sure I can see how LookupHost would return scoped IPs